### PR TITLE
Adding user admin test cases

### PR DESCRIPTION
**Pull request by thusithathilina**

### CODE REVIEW

File: modules/integration/tests-integration/tests/src/test/java/org/wso2/appserver/integration/tests/usermgt/UserManagementWithAdminUserTestCase.java [hunk 1]
```diff
@@ -0,0 +1,408 @@
...
+				isAssignedRolesAvailable &= flaggedName.getSelected();
+				continue;
+			}else if ("Internal/everyone".equals(flaggedName.getItemName())) {
+				assertTrue(flaggedName.getSelected(), "User don't have Internal/everyone role");
+				isAssignedRolesAvailable &= flaggedName.getSelected();
+				continue;
+			}
+			isOtherRolesAvailable |= flaggedName.getSelected();
+		}
+		if(isOtherRolesAvailable || !isAssignedRolesAvailable) {
```
Comment (Nipuni) [inline modules/integration/tests-integration/tests/src/test/java/org/wso2/appserver/integration/tests/usermgt/UserManagementWithAdminUserTestCase.java]: formatting issue in the if condition

File: modules/integration/tests-integration/tests/src/test/java/org/wso2/appserver/integration/tests/usermgt/UserManagementWithAdminUserTestCase.java [hunk 2]
```diff
@@ -0,0 +1,408 @@
...
+		FlaggedName[] listOfUsers = userManagementClient.listAllUsers("*", 10);
+		// Get user count defined in automation.xml + last element is false when call through the UserAdminClient
+		if (userMode == TestUserMode.SUPER_TENANT_ADMIN) {
+			assertTrue(listOfUsers.length >= 6, "User count is differ from the expected count");
+		} else {
+			assertTrue(listOfUsers.length >= 5, "User count is differ from the expected count");
+		}
+	}
+
+	@Test(groups = "wso2.as", description = "Change password of current user defined in automation.xml")
```
Comment (Niranjan-K) [inline modules/integration/tests-integration/tests/src/test/java/org/wso2/appserver/integration/tests/usermgt/UserManagementWithAdminUserTestCase.java]: If a method in a test class as a depends on then as a practice need to define depends on for all other methods.

File: modules/integration/tests-integration/tests/src/test/java/org/wso2/appserver/integration/tests/usermgt/UserManagementWithAdminUserTestCase.java [hunk 3]
```diff
@@ -0,0 +1,408 @@
...
+		assertEquals(ex.getMessage(), "Error while login as " + username,
+		             "User able to login to system using old password");
+
+		if (userMode == TestUserMode.TENANT_ADMIN) {
+			username = oldUsername;
+		}
+		password = newPassword;
+	}
+
+	@Test(groups = "wso2.as", description = "Deleting a user", dependsOnMethods = { "testAddRemoveUsersOfRole" })
```
Comment (Niranjan-K) [inline modules/integration/tests-integration/tests/src/test/java/org/wso2/appserver/integration/tests/usermgt/UserManagementWithAdminUserTestCase.java]: If a method in a test class as a depends on then as a practice need to define depends on for all other methods.

File: modules/integration/tests-integration/tests/src/test/java/org/wso2/appserver/integration/tests/usermgt/UserManagementWithAdminUserTestCase.java [hunk 4]
```diff
@@ -0,0 +1,408 @@
...
+		Path filePath = Paths.get(FrameworkPathUtil.getSystemResourceLocation(), "artifacts", "AS", "usermgt",
+		                          USER_CSV_FILE_NAME);
+		DataHandler handler = new DataHandler(filePath.toUri().toURL());
+		userManagementClient.bulkImportUsers(filePath.toString(), handler, "abc123");
+		ArrayList<String> users = (ArrayList<String>) FileUtils.readLines(filePath.toFile());
+		users.remove(0); // Remove the username
+		HashSet<String> userList = userManagementClient.getUserList();
+		for (String user : users) {
+			assertTrue(userList.contains(user), "Username " + user + " doesn't exist");
+			//assertNotNull(loginLogoutClient.login(user, "abc123", asServer.getInstance().getHosts().get("default")));
```
Comment (callkalpa) [inline modules/integration/tests-integration/tests/src/test/java/org/wso2/appserver/integration/tests/usermgt/UserManagementWithAdminUserTestCase.java]: Why commented?

File: modules/integration/tests-integration/tests/src/test/java/org/wso2/appserver/integration/tests/usermgt/UserManagementWithNonAdminUserTestCase.java [hunk 5]
```diff
@@ -0,0 +1,278 @@
+package org.wso2.appserver.integration.tests.usermgt;
+/*
+* Copyright (c) 2014, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
```
Comment (callkalpa) [inline modules/integration/tests-integration/tests/src/test/java/org/wso2/appserver/integration/tests/usermgt/UserManagementWithNonAdminUserTestCase.java]: Update year

File: modules/integration/tests-integration/tests/src/test/java/org/wso2/appserver/integration/tests/usermgt/UserManagementWithNonAdminUserTestCase.java [hunk 6]
```diff
@@ -0,0 +1,278 @@
...
+	@Factory(dataProvider = "userModeDataProvider")
+	public UserManagementWithNonAdminUserTestCase(TestUserMode userMode) {
+		this.userMode = userMode;
+	}
+
+	@BeforeClass(alwaysRun = true)
+	public void setEnvironment() throws Exception {
+		//Since if we don't impose roles in the automation.xml all the users have admin privileges created 2 new users and a role and use that for TENANT and SUPER_TENANT users
+		if (userMode == TestUserMode.SUPER_TENANT_USER) {
+
```
Comment (callkalpa) [inline modules/integration/tests-integration/tests/src/test/java/org/wso2/appserver/integration/tests/usermgt/UserManagementWithNonAdminUserTestCase.java]: additional blank line?

File: modules/integration/tests-integration/tests/src/test/java/org/wso2/appserver/integration/tests/usermgt/UserManagementWithNonAdminUserTestCase.java [hunk 7]
```diff
@@ -0,0 +1,278 @@
...
+	}
+
+	@Test(groups = "wso2.as", description = "Get all shared role names", expectedExceptions = RemoteException.class,
+			expectedExceptionsMessageRegExp = NON_ADMIN_EXCEPTION_MESSAGE)
+	public void testGetAllSharedRoles() throws Exception {
+		userManagementClient.getAllSharedRoleNames("*", 10);
+		fail("Non admin user able to get all shared roles");
+	}
+
+	//users who have login permission is able to execute this.
```
Comment (callkalpa) [inline modules/integration/tests-integration/tests/src/test/java/org/wso2/appserver/integration/tests/usermgt/UserManagementWithNonAdminUserTestCase.java]: If not used, shall we remove the method?

File: modules/integration/tests-integration/tests/src/test/java/org/wso2/appserver/integration/tests/usermgt/UserManagementWithAdminUserTestCase.java [hunk 8]
```diff
@@ -0,0 +1,408 @@
...
+				.updateRolesOfUser(userName, new String[] { TEST_NON_ADMIN_ROLE_NAME, "Internal/everyone" });
+		FlaggedName[] rolesOfUser = userManagementClient.getRolesOfUser(userName, "*", 10);
+		boolean isOtherRolesAvailable = false;
+		boolean isAssignedRolesAvailable = true;
+		for (FlaggedName flaggedName : rolesOfUser) {
+			if (TEST_NON_ADMIN_ROLE_NAME.equals(flaggedName.getItemName())) {
+				assertTrue(flaggedName.getSelected(), "Updating roles of users failed");
+				isAssignedRolesAvailable &= flaggedName.getSelected();
+				continue;
+			}else if ("Internal/everyone".equals(flaggedName.getItemName())) {
```
Comment (arunasujith) [inline modules/integration/tests-integration/tests/src/test/java/org/wso2/appserver/integration/tests/usermgt/UserManagementWithAdminUserTestCase.java]: formatting issue

**Pull request merged**